### ISSUE TITLE
Latest Fabric 1.20.1 Version Potentially Missing Fixes

### ISSUE THREAD

**Issue by RPGNinja** [Apollounknowndev/tectonic#217]
Hello,

This might be a me issue rather than something wrong with the mod but I couldn't find a different way to reach out other than here.

From my testing on both my modpack and a fresh instance with just Tectonic + Terralith, it seems like the fixes for the world height and Terralith's mega caves aren't actually working (haven't tested any of the other changes).

After changing the setting in the config to allow for the increased height, the world height is still stuck at 319 and no Terralith mega caves are visible/spawning.

On a fresh instance, I tested Tectonic on its own and was able to build past 319, reaching up to 639. 
This leads me to believe that the issue hasn't been resolved, or something went wrong.

I also have 2 questions about the mod if you don't mind :)

Since Terratonic isn't updated yet (which I assume that means that it also has the same problem), is it possible to turn Tectonic into a datapack by removing meta-inf and changing it to a .zip, or would that mess something up?

Also, I've been having an issue where structures are spawning incorrectly due to Tectonic (at least from my testing).
For example, a structure meant for the bottom of the ocean is spawning on a beach or mineshafts are spawning in the air/connected to a mountain top rather than being submerged into the ground.
Would you happen to have any advice on how I can at least reduce the chances of this happening?

Thank you very much for your time!
![2024-09-02_23 39 07](https://github.com/user-attachments/assets/7a70dfff-4925-4c6c-ac62-63c066fb5f1e)
![2024-08-06_00 02 44](https://github.com/user-attachments/assets/1a59abbe-03be-4d9b-8de1-f17038fc813a)

**Comment by Kyles1201** [comment]
Yes I am experiencing the same issues. No visible mega caves, and still some haze mountains areas are flat. It almost seems the fixes in the latest version didn't happen at all.

**Comment by RPGNinja** [comment]
I COULD be wrong but after disabling Terralith, it seems like my worldgen has done a complete 180.
Biomes look way better now, they're much larger, I'm not seeing any micro biomes or snow biomes spawning in a desert.
The biggest thing is structures are generating normal again (as far as I've been able to see) as well as being able to increase the world height for tectonic.
Could it be possible that there's an unknown issue between Tectonic and Terralith?

**Comment by KB-Q** [comment]
Same issues: still stuck with world height 319, and mountains are getting cut off at 320 with increased_height set to true.

**Comment by Apollounknowndev** [comment]
Going to split this into two to better handle the issues.
#219 will be for mountains cutting off above y320
#217 (this one) will be for broken structure generation.

**Comment by RPGNinja** [comment]
Thank you for your time and help!

For some more context, 1 of the mods that is breaking majority of the time with Tectonic (default config and experimental) is Hopo's better mineshafts.
Other than that, it's mostly been the occasional floating structures, structures digging out terrain (this is probably normal) and structures spawning in odd places.
Let me know if there's any more information you need from me!

**Comment by spencerdfhyfdjdn** [comment]
This still happens a lot, please fix, its been almost 2 years
